src/fleet_lock/mod.rs

```diff
@@ -0,0 +1,259 @@
+//! Asynchronous FleetLock client, remote lock management.
+//!
+//! This module implements a client for FleetLock, a bare HTTP
+//! protocol for managing cluster-wide reboot via a remote
+//! lock manager. Protocol specification is currently in progress at
+//! https://github.com/coreos/airlock/pull/1.
+
+use crate::identity::Identity;
+use failure::{Error, Fail, Fallible, ResultExt};
+use futures::future;
+use futures::prelude::*;
+use reqwest::r#async as asynchro;
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)]
+mod mock_tests;
+
+/// FleetLock pre-reboot API path endpoint (v1).
+static V1_PRE_REBOOT: &str = "v1/pre-reboot";
+
+/// FleetLock steady-state API path endpoint (v1).
+static V1_STEADY_STATE: &str = "v1/steady-state";
+
+/// Error from lock manager.
+#[derive(Clone, Debug, Fail, Deserialize, Serialize, PartialEq, Eq)]
+pub struct LockRejection {
+    /// Endpoint that returned this rejection/error.
+    #[serde(skip)]
+    endpoint: String,
+    /// HTTP status code returned by the server.
+    #[serde(skip)]
+    status: reqwest::StatusCode,
+    /// Machine-friendly brief error kind.
+    kind: String,
+    /// Human-friendly detailed error explanation.
+    value: String,
+}
+
+impl std::fmt::Display for LockRejection {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "lock manager {} rejection, code {}: {}",
+            self.endpoint,
+            self.status.as_u16(),
+            self.value
+        )
+    }
+}
+
+impl LockRejection {
+    /// Convert a response into a lock result.
+    fn error_for_response(
```

Comment:
This name honestly confused me for a bit because I didn't realized it also handled the happy case. I think it'd be easier to understand as a stand-alone function named e.g. `map_response()` or something?

Reply:
You are right, this is overly confusing to follow.
Let me see if I can actually manage to scope this back to the error-path only, which I think would make this even easier to read. If that is not feasible for some reason, I'll rename as suggested,